### PR TITLE
Merge xcconfig from inherited targets

### PR DESCRIPTION
**Pull request by dnkoutso**
Closes https://github.com/CocoaPods/CocoaPods/issues/8033
Closes https://github.com/CocoaPods/CocoaPods/issues/8497

### CODE REVIEW

**Comment by CocoaPodsBarista** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please include a CHANGELOG entry to credit yourself! <br />
You can find it at <a href="https://github.com/CocoaPods/CocoaPods/blob/master/CHANGELOG.md">CHANGELOG.md</a>.</td>
    </tr>
  </tbody>
</table>

Here's an example of your CHANGELOG entry:

```markdown
* Merge xcconfig from inherited targets  
  [dnkoutso](https://github.com/dnkoutso)
  [#issue_number](https://github.com/CocoaPods/CocoaPods/issues/issue_number)
```

*note*: There are two invisible spaces after the entry's text.


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: spec/unit/target/build_settings/aggregate_target_settings_spec.rb [hunk 1]
```diff
@@ -713,7 +713,7 @@ def specs
               @target.pod_targets.each { |pt| pt.spec_consumers.each { |sc| sc.stubs(:frameworks => %w(UIKit), :libraries => %w(z c++)) } }
 
               @xcconfig = @generator.generate
-              @xcconfig.to_hash['OTHER_LDFLAGS'].should == '$(inherited) -ObjC -l"c++" -l"z" -framework "UIKit"'
+              @xcconfig.to_hash['OTHER_LDFLAGS'].should == '$(inherited) -ObjC -no_compact_unwind -l"c++" -l"z" -framework "UIKit"'
```
Comment (amorde) [inline spec/unit/target/build_settings/aggregate_target_settings_spec.rb]: where is this flag coming from? it's unclear how this shows that the issue is fixed, though I haven't looked at the fixtures for this

Comment (dnkoutso) [reply spec/unit/target/build_settings/aggregate_target_settings_spec.rb]: From the tests:
```ruby        
before do
          @target_definition = fixture_target_definition
          @specs = specs
          @specs.first.user_target_xcconfig = { 'OTHER_LDFLAGS' => '-no_compact_unwind', 'USE_HEADERMAP' => 'NO' } unless @specs.empty?
          @specs.first.pod_target_xcconfig = { 'CLANG_CXX_LANGUAGE_STANDARD' => 'c++11' } unless @specs.empty?
          @pod_targets = @specs.map { |spec| pod_target(spec, @target_definition) }
          @target = fixture_aggregate_target(@pod_targets, false, { 'Release' => :release }, [], Platform.new(:ios, '6.0'), @target_definition)
          unless @specs.empty?
            @target.target_definition.whitelist_pod_for_configuration(@specs.first.name, 'Release')
          end
          @generator = AggregateTargetSettings.new(@target, 'Release')
        end
```

Comment (segiddins) [reply spec/unit/target/build_settings/aggregate_target_settings_spec.rb]: it'd be great if you could add a more explicit test, so a future cleanup doesn't accidentally regress behavior again

Comment (dnkoutso) [reply spec/unit/target/build_settings/aggregate_target_settings_spec.rb]: Will do.

**Comment by nateuni** [comment]
How do I get this changes?

**Comment by amorde** [comment]
@nateuni this is an open pull request, so other than checking out the branch directly this isn't yet available to use. Please be patient & we will release this as soon as possible :)

**Comment by dnkoutso** [comment]
Tried to add a test for this but it seems like its actually already covered by what we have and I Had to update here. The target includes the search path aggregate targets and the setup is meant to include it.

Please push back if you feel like I should spend more time to add one here but I do believe its covered.

**Pull request merged**